src/Compression/ICompressionCodec.h

```diff
@@ -45,8 +45,21 @@ class ICompressionCodec : private boost::noncopyable
     /// Compressed bytes from uncompressed source to dest. Dest should preallocate memory
     UInt32 compress(const char * source, UInt32 source_size, char * dest) const;
 
-    /// Decompress bytes from compressed source to dest. Dest should preallocate memory
-    UInt32 decompress(const char * source, UInt32 source_size, char * dest) const;
+    /// Decompress bytes from compressed source to dest. Dest should preallocate memory;
+    // reqType is specific for HW decompressor:
+    //0 means synchronous request by default;
+    //1 means asynchronous request, must be used in pair with decompressFlush;
+    //2 means SW decompressor instead of HW
+    UInt32 decompress(const char * source, UInt32 source_size, char * dest, UInt8 req_type = 0);
+
+    /// Flush all asynchronous request for decompression
+    void decompressFlush(void);
```

Comment:
I've add sentences as below into comment:

Flush result for previous asynchronous decompression requests.
This function must be called following several requests offload to HW, 
To make sure asynchronous results have been flushed into target buffer completely.
Meanwhile, source and target buffer for decompression should not be overwritten until this function execute completely. 
Otherwise it would conflict with HW offloading and cause exception.
For QPL deflate, it support the maximum number of requests equal to DeflateJobHWPool::jobPoolSize